lib/elixir/lib/list.ex

```diff
@@ -591,15 +591,21 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
-      result when is_binary(result) ->
-        result
-
-      {:error, encoded, rest} ->
-        raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :invalid
-
-      {:incomplete, encoded, rest} ->
-        raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :incomplete
+    try do
+      case :unicode.characters_to_binary(list) do
+        result when is_binary(result) ->
+          result
+
+        {:error, encoded, rest} ->
+          raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :invalid
+
+        {:incomplete, encoded, rest} ->
+          raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :incomplete
+      end
+    rescue
+      ArgumentError ->
+        # Instead of a more cryptic ":unicode.characters_to_binary" error.
```

Comment:
I think we can avoid this comment by rewriting `try` clause like this:
```elixir
try do
  :unicode.characters_to_binary(list)
rescue
  ArgumentError ->
    # Instead of a more cryptic ":unicode.characters_to_binary" error.
    raise ArgumentError, "cannot make this list into a string (use 'inspect(list)' for its internal representation as a string)"
else
  result when is_binary(result) ->
    result

  {:error, encoded, rest} ->
    raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :invalid

  {:incomplete, encoded, rest} ->
    raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :incomplete
end
```
it clearly shows where we expect a failure.